internal/sourcer/registry_sourcer.go

```diff
@@ -53,17 +55,23 @@ func (cs catalogSource) Source(ctx context.Context, po *platformv1alpha1.Platfor
 	return latestBundle, nil
 }
 
-func (s sources) GetCandidates(ctx context.Context, po *platformv1alpha1.PlatformOperator) (bundles, error) {
+func (s sources) GetCandidates(ctx context.Context, po *platformv1alpha1.PlatformOperator, runLocal bool) (bundles, error) {
+	// TODO(tflannag): This doesn't account for edge case where there are zero sources.
+	// TODO(tflannag): Clean up the run local implementation
 	if len(s) != 1 {
 		return nil, fmt.Errorf("validation error: only a single catalog source is supported during phase 0")
 	}
 	cs := s[0]
 
+	address := cs.Status.GRPCConnectionState.Address
+	if runLocal {
+		address = "localhost:50051"
+	}
```

Comment:
Non-blocking nit: I've seen issues crop up before with `run-local` style flags like this because its always a proxy flag for what someone is really trying to do. In this case, it seems to be overriding the catSrc address. Two alternatives would be:
1. Creating a catalog source in the cluster that actually uses `localhost:50051` as the address, thus no client-specific changes necessary (I assume)
2. Change the flag name to `catalog-source-address-override` (or similar) and then pass `localhost:50051` in via CLI when running locally, thus not really a "local" vs "cluster" thing -- more of a general configuration thing.

Reply:
I'd also consider putting a `getAddress func(cs *operatorsv1alpha1.CatalogSource) string` func field on the `catalogSource` struct, calling that function here, and then defining different constructors to handle the override or not.

Reply:
This was a debug artifact more-or-less that I forgot to cut in the latest rounds of edits. I think I'd need to just refactor the Source implementation to make it easier to mock catalog contents but I haven't gotten around to that yet. Most of the current e2e tests that got introduced here could live as unit tests once that behavior is easier to mock.